lib/Sema/CSApply.cpp

```diff
@@ -9287,6 +9288,55 @@ static llvm::Optional<SyntacticElementTarget> applySolutionToForEachStmt(
            "Couldn't find sequence conformance");
     stmt->setSequenceConformance(sequenceConformance);
   }
+  
+  auto &ctx = cs.getASTContext();
+  if (ctx.LangOpts.hasFeature(Feature::PackIteration)) {
+    if (forEachStmtInfo.isa<PackIterationInfo>()) {
+      auto packIterationInfo = *forEachStmtInfo.dyn_cast<PackIterationInfo>();
```

Comment:
Use `if (auto *<<option>> = <<union>>.dyn_cast<...>()) {` instead of `if (<<union>>.is<...>()) { auto <<option>> = <<union>>.dyn_cast<...>(); }`, another plot like this - https://github.com/apple/swift/pull/67594/files#diff-1ba7c144bdfaf2daf57703f890ef9575a5782850e77c573df4170a547d73ab99R9135